File: Data Structures & Algorithms/balanced-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    boolean isBal = true;
    public boolean isBalanced(TreeNode root) {
        depth(root);
        return isBal;

    }

    public int depth(TreeNode root){
        if(root == null) return 0;
        int a = depth(root.left);
        int b = depth(root.right);
        if(Math.abs(a - b) > 1) isBal = false;
        if(a > b) return a+1;
        return b+1;
    }
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    vector<int> a;
public:
    vector<int> inorderTraversal(TreeNode* root) {
        tr(root);
        return a;
    }


    void tr(TreeNode* root){
        if(root == NULL) return;
        if(root->left){
            inorderTraversal(root->left);
        }
        a.push_back(root->val);
        if(root->right){
            inorderTraversal(root->right);
        }
    }
};